File: news/src/main/java/com/leory/badminton/news/di/component/MatchDateComponent.java
package com.leory.badminton.news.di.component;

import com.leory.badminton.news.mvp.contract.MatchDetailContract;
import com.leory.badminton.news.mvp.model.bean.MatchTabDateBean;
import com.leory.badminton.news.mvp.ui.fragment.MatchDateFragment;
import com.leory.commonlib.di.scope.FragmentScope;

import java.util.List;

import dagger.BindsInstance;
import dagger.Subcomponent;

/**
 * Describe : 赛程component
 * Author : leory
 * Date : 2019-06-06
 */
@Subcomponent
@FragmentScope
public interface MatchDateComponent {
    void inject(MatchDateFragment fragment);

    @Subcomponent.Builder
    interface Builder {
        @BindsInstance
        MatchDateComponent.Builder view(MatchDetailContract.MatchDateView view);
        @BindsInstance
        MatchDateComponent.Builder tabDates(List<MatchTabDateBean>tabDates);
        MatchDateComponent build();
    }
}

File: news/src/main/java/com/leory/badminton/news/di/component/MatchDateComponent.kt
package com.leory.badminton.news.di.component

import com.leory.badminton.news.mvp.contract.MatchDetailContract
import com.leory.badminton.news.mvp.model.bean.MatchTabDateBean
import com.leory.badminton.news.mvp.ui.fragment.MatchDateFragment
import com.leory.commonlib.di.scope.FragmentScope
import dagger.BindsInstance
import dagger.Subcomponent
import javax.inject.Named

/**
 * Describe : 赛程component
 * Author : leory
 * Date : 2019-06-06
 */
@Subcomponent
@FragmentScope
interface MatchDateComponent {
    fun inject(fragment: MatchDateFragment)

    @Subcomponent.Builder
    interface Builder {
        @BindsInstance
        fun view(view: MatchDetailContract.MatchDateView): Builder

        @BindsInstance
        fun tabDates(tabDates: MutableList<MatchTabDateBean>): Builder

        @BindsInstance
        fun country(@Named("country") country: String): Builder

        fun build(): MatchDateComponent
    }
}
